src/strings_func.h

```diff
@@ -79,37 +81,55 @@ inline int64_t PackVelocity(uint speed, VehicleType type)
 	return speed | (static_cast<uint64_t>(type) << 56);
 }
 
-void SetDParam(size_t n, uint64_t v);
-void SetDParamMaxValue(size_t n, uint64_t max_value, uint min_count = 0, FontSize size = FS_NORMAL);
-void SetDParamMaxDigits(size_t n, uint count, FontSize size = FS_NORMAL);
+uint64_t GetParamMaxValue(uint64_t max_value, uint min_count = 0, FontSize size = FS_NORMAL);
+uint64_t GetParamMaxDigits(uint count, FontSize size = FS_NORMAL);
 
-template <typename T, std::enable_if_t<std::is_base_of<StrongTypedefBase, T>::value, int> = 0>
-void SetDParam(size_t n, T v)
+extern TextDirection _current_text_dir; ///< Text direction of the currently selected language
+
+void InitializeLanguagePacks();
+const char *GetCurrentLanguageIsoCode();
+std::string_view GetListSeparator();
+
+/**
+ * Helper to create the StringParameters with its own buffer with the given
+ * parameter values.
+ * @param args The parameters to set for the to be created StringParameters.
+ * @return The constructed StringParameters.
+ */
+template <typename... Args>
+auto MakeParameters(const Args&... args)
 {
-	SetDParam(n, v.base());
+	return std::array<StringParameter, sizeof...(args)>({args...});
```

Comment:
No idea... but might this solve the problem on MSVC?
```suggestion
        return std::array<StringParameter, sizeof...(args)>({(StringParameter(args), ...)});
```